File: MainProject/src/ERP2/Controller.java
package ERP2;

import java.rmi.RemoteException;

import javax.swing.table.DefaultTableModel;

import org.tempuri.ERP2_WebServiceSoapProxy;

public class Controller {
    
    private org.tempuri.ERP2_WebServiceSoapProxy client = new org.tempuri.ERP2_WebServiceSoapProxy();
    ExceptionHandler exceptionHandler = new ExceptionHandler();
    
    //DATA METHODS
    
    public DefaultTableModel employeeContents(String args, DefaultTableModel tableModel){
        try {
            return Utils.convertListToTable(client.employeeContents(args), tableModel);
        }
        catch(RemoteException rE) {
            exceptionHandler.exceptionMessage(rE);
            return null;
        }
    }
    
    public DefaultTableModel employeeRelatives(DefaultTableModel tableModel) {
        try {
            return Utils.convertListToTable(client.employeeRelatives(), tableModel);
        }
        catch (RemoteException rE) {
            exceptionHandler.exceptionMessage(rE);
            return null;
        }
    }
    
    public DefaultTableModel sickEmployees(DefaultTableModel tableModel) {
        try {
            return Utils.convertListToTable(client.sickEmployees(), tableModel);
        }
        catch (RemoteException rE) {
            exceptionHandler.exceptionMessage(rE);
            return null;
        }
    }
    
    public DefaultTableModel mostSickEmployee(DefaultTableModel tableModel) {
        try {
            return Utils.convertListToTable(client.mostSickEmployee(), tableModel);
        }
        catch (RemoteException rE) {
            exceptionHandler.exceptionMessage(rE);
            return null;
        }
    }
    
    public DefaultTableModel getAllKeys(DefaultTableModel tableModel, String args) {
        try {
            return Utils.convertListToTable(client.getAllKeys(args), tableModel);
        }
        catch (RemoteException rE) {
            exceptionHandler.exceptionMessage(rE);
            return null;
        }
    }
    
    //METADATA METHODS
    
    public DefaultTableModel allIndexes (DefaultTableModel tableModel, String args) {
        try {
            return Utils.convertListToTable(client.allIndexes(args), tableModel);
        }
        catch (RemoteException rE) {
            exceptionHandler.exceptionMessage(rE);
            return null;
        }
    }
    
    public DefaultTableModel allConstraints (DefaultTableModel tableModel, String args) {
        try {
            return Utils.convertListToTable(client.allConstraints(args), tableModel);
        }
        catch (RemoteException rE) {
            exceptionHandler.exceptionMessage(rE);
            return null;
        }
    }
    
    public DefaultTableModel allTableNames1 (DefaultTableModel tableModel) {
        try {
            return Utils.convertListToTable(client.allTableNames1(), tableModel);
        }
        catch (RemoteException rE) {
            exceptionHandler.exceptionMessage(rE);
            return null;
        }
    }
    
    public DefaultTableModel allTableNames2 (DefaultTableModel tableModel) {
        try {
            return Utils.convertListToTable(client.allTableNames2(), tableModel);
        }
        catch (RemoteException rE) {
            exceptionHandler.exceptionMessage(rE);
            return null;
        }
    }
    
    public DefaultTableModel allEmployeeColumns1 (DefaultTableModel tableModel) {
        try {
            return Utils.convertListToTable(client.allEmployeeColumns1(), tableModel);
        }
        catch (RemoteException rE) {
            exceptionHandler.exceptionMessage(rE);
            return null;
        }
    }
    
    public DefaultTableModel allEmployeeColumns2 (DefaultTableModel tableModel) {
        try {
            return Utils.convertListToTable(client.allEmployeeColumns2(), tableModel);
        }
        catch (RemoteException rE) {
            exceptionHandler.exceptionMessage(rE);
            return null;
        }
    }
    
    //HEJ
    
    // ALL EMPLOYEE TABLE'S TABLE NAMES
    public String[] allEmployeeTables() {
        try {
            return client.employeeTablesTableNames();
        }
        catch (RemoteException rE) {
            exceptionHandler.exceptionMessage(rE);
            return null;
        }
    }
}

File: SharpeDiem/ERP2_WindowsForm/Controller.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ERP2_WindowsForm
{
    public class Controller
    {
        WebServiceClient.ERP2_WebServiceSoapClient client; 

        public Controller()
        {
           client = new WebServiceClient.ERP2_WebServiceSoapClient();
        }

        //Data methods

        public DataTable EmployeeContents(string args)
        {
            return Utils.ConvertListToDataTable(client.EmployeeContents(args));
        }
        public DataTable EmployeeRelatives()
        {
            return Utils.ConvertListToDataTable(client.EmployeeRelatives());
        }

        public DataTable SickEmployees()
        {
            return Utils.ConvertListToDataTable(client.SickEmployees());
        }

        public DataTable MostSickEmployee()
        {
            return Utils.ConvertListToDataTable(client.MostSickEmployee());
        }

        //Metadata methods
        public DataTable GetAllKeys(string args)
        {
            return Utils.ConvertListToDataTable(client.GetAllKeys(args));                
        }

        public DataTable AllIndexes(string args)
        {
            return Utils.ConvertListToDataTable(client.AllIndexes(args));
        }

        public DataTable AllConstraints(string args)
        {
            return Utils.ConvertListToDataTable(client.AllConstraints(args));
        }

        public DataTable AllTableNames1()
        {
            return Utils.ConvertListToDataTable(client.AllTableNames1());
        }

        public DataTable AllTableNames2()
        {
            return Utils.ConvertListToDataTable(client.AllTableNames2());
        }

        public DataTable AllEmployeeColumnNames1()
        {
            return Utils.ConvertListToDataTable(client.AllEmployeeColumns1());
        }

        public DataTable AllEmployeeColumns2()
        {
            return Utils.ConvertListToDataTable(client.AllEmployeeColumns2());
        }

        public object[] EmployeeTablesTableNames()
        {
            object[] tableNames = new object[7];
            List<string> stringTableNames = client.EmployeeTablesTableNames();
            int i = 0;
            foreach (string table in stringTableNames)
            {
                tableNames[i] = table;
                i++;
            }
            return tableNames;
        }

        
    }
}
